src/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/ExternalReferencesTable.cs

```diff
@@ -68,26 +68,17 @@ public bool InitializeCommonFixupsTable(IntPtr moduleHandle)
 
         unsafe public uint GetRvaFromIndex(uint index)
         {
+            throw new PlatformNotSupportedException();
+        }
+
+        unsafe public IntPtr GetIntPtrFromIndex(uint index)
+        {
             Debug.Assert(_moduleHandle != IntPtr.Zero);
 
             if (index >= _RVAsCount)
                 throw new BadImageFormatException();
 
-            return ((uint*)_RVAs)[index];
-        }
-
-        unsafe public IntPtr GetIntPtrFromIndex(uint index)
-        {
-            uint rva = GetRvaFromIndex(index);
-            if ((rva & 0x80000000) != 0)
```

Comment:
REL32 should be the efficient portable-enough story (BTW: I saw Hongtao using it for the interface work that he is doing.)

We should have a flag in the binary somewhere that says how image pointers are encoded. We should be able to handle at least full pointers (less efficient, but most portable solution - e.g. for CppCodeGen) and REL32 (more efficient, portable enough). Switch in this method based on what the flag says.